### ISSUE TITLE
Inspection for possible unhandled error in event handler

### ISSUE THREAD

**Issue by retailcoder** [rubberduck-vba/Rubberduck#3549]
Consider `Class1`:

```vb

Option Explicit
Public Event Foo()

Public Sub DoSomething()
    On Error GoTo AhHa
    RaiseEvent Foo
    Exit Sub
AhHa:
    MsgBox "Ah-HA!"
End Sub
```

And then, say, `ThisWorkbook` (makes a convenient class to have a `WithEvents` field in):

```vb
Option Explicit
Private WithEvents bar As Class1

Private Sub bar_Foo()
    Err.Raise 5
End Sub

Public Sub Test()
    Set bar = New Class1
    bar.DoSomething
End Sub
```

Despite the `AhHa` error handler, the message box will never be shown, and the error raised in the `bar_Foo()` handler is unhandled.

We could issue a warning-level (error-level?) inspection result when a procedure that contains a `RaiseEvent` statement also contains an `On Error` statement, saying something like:

> **Possible unhandled run-time error in event handler**
>
> A 'RaiseEvent' statement appears to be covered by an 'On Error' statement, but because a VBA event handler runs on its own call stack, an error raised in an event handler would not bubble back up to the event source, leaving the error unhandled, which may be unexpected behavior.

A quickfix could be a refactoring that adds a `ByRef errDescription As String` parameter to the event signature, its `RaiseEvent` invocations, and handler signatures, adds an `On Error GoTo BubbleItUp` statement and a corresponding label (hmm, prompt for a label identifier name?), and the error-handling subroutine would assign the `errDescription` parameter. If there's already an `On Error GoTo` statement in the handler, then we could just insert a `errDescription = Err.Description` statement on the first line after the label. If there's more than one `On Error GoTo` statement, we can just leave it as-is and let the user sort it out.

**Comment by bclothier** [comment]
Just to point out that it might be totally legit to have error handling within the procedure that does `RaiseEvent` to handle errors not needing to do with the `RaiseEvent` (e.g. suppose an event has parameters, and in process of filling in the data for parameters, something went horribly wrong, so error handling is needed in the procedure, which might also means the `RaiseEvent` may not happen. 

IDK if it's more common to have a potentially erring procedure with a `RaiseEvent` that doesn't give a hoot what happened in the subscribers than it is to have a procedure that really hopes to catch errors from subscribers (which will fail miserably as demonstrated above). 

I prefer to show that at all time, just to remind of possible holes; I don't know if it's possible to catch the distinctions between 2 cases but just in case, it might be nice to reduce # of false positives.

**Comment by retailcoder** [comment]
@bclothier you're right. Actually I think the only sensible thing to warn about, might be the *absence* of an `On Error` statement in an event handler, regardless of who's firing the event.
